controllers/cluster_controller.go

```diff
@@ -287,3 +296,57 @@ func splitMachineList(list *clusterv1.MachineList) (*clusterv1.MachineList, *clu
 	}
 	return controlplanes, nodes
 }
+
+func (r *ClusterReconciler) reconcileClusterStatus(ctx context.Context, cluster *clusterv1.Cluster) error {
+	if cluster.Status.ControlPlaneInitialized {
+		return nil
+	}
+
+	machines := &clusterv1.MachineList{}
+	err := r.Client.List(ctx, machines, client.InNamespace(cluster.Namespace))
+	if err != nil {
+		return err
+	}
+
+	for _, m := range machines.Items {
+		if util.IsControlPlaneMachine(&m) && m.Status.NodeRef != nil {
+			cluster.Status.ControlPlaneInitialized = true
+			return nil
+		}
+	}
+
+	return nil
+}
+
+// MachineToMachineSets is a handler.ToRequestsFunc to be used to enqeue requests for reconciliation
+// for Cluster to update its status.controlplaneInitialized field
+func (r *ClusterReconciler) MachineToCluster(o handler.MapObject) []ctrl.Request {
+	result := []ctrl.Request{}
+
+	m, ok := o.Object.(*clusterv1.Machine)
+	if !ok {
+		r.Log.Error(errors.New("did not get machine"), "got", fmt.Sprintf("%T", o.Object))
+		return result
+	}
+
+	cluster, err := util.GetClusterFromMetadata(context.Background(), r.Client, m.ObjectMeta)
+	if err != nil {
+		r.Log.Error(err, "failed to get cluster", "machine", m.Name, "cluster", m.ClusterName)
+		return result
+	}
+
+	if cluster.Status.ControlPlaneInitialized {
+		return result
```

Comment:
Can we return `nil` instead of the empty result in any case except for when we initialize the ctrl.Request?